src/custom_nodes/east_ocr/nms.hpp

```diff
@@ -93,21 +93,26 @@ inline void nms(const std::vector<cv::Rect>& srcRects,
  * @param thresh
  * @param neighbors
  */
+template <typename T>
 inline void nms2(const std::vector<cv::Rect>& srcRects,
     const std::vector<float>& scores,
+    const std::vector<T>& metadata,
     std::vector<cv::Rect>& resRects,
     std::vector<float>& resScores,
+    std::vector<T>& resMetadata,
     float thresh,
     int neighbors = 0,
     float minScoresSum = 0.f) {
     resRects.clear();
     resScores.clear();
+    resMetadata.clear();
```

Comment:
Why we have to clear incoming vectors?

Reply:
To make sure caller did not send populated vector. I tried not to modify nms.hpp a lot since this is external (https://github.com/Nuzhny007/Non-Maximum-Suppression/blob/master/nms.h)

Reply:
Seems like overkill but ok.